classes/Renderer/User.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace MySociety\TheyWorkForYou\Renderer;
+
+/**
+ * User data for headers
+ */
+
+class User
+{
+    public $data;
+
+    private $header;
+    private $user;
+    private $pagedata;
+    private $page;
+    private $returl;
+
+    public function __construct($header) {
+        global $THEUSER, $DATA, $this_page;
+
+        $this->header = $header;
+        $this->user = $THEUSER;
+        $this->pagedata = $DATA;
+        $this->page = $this_page;
+        $this->data = array();
+        $this->setupNavLinks();
+        $this->addRepLinks();
+    }
+
+    private function setupNavLinks() {
+
+        $this->data['user_nav_links'] = array();
+
+        // We may want to send the user back to this current page after they've
+        // joined, logged out or logged in. So we put the URL in $returl.
+        $URL = new \URL($this->page);
+        $this->returl = $URL->generate('none');
+
+        //user logged in
+        if ($this->user->isloggedin()) {
+            $this->addLoggedInLinks();
+        } else {
+            $this->addLoggedOutLinks();
+        }
+    }
+
+    private function addLoggedInLinks() {
+        // The 'Edit details' link.
+        $menudata   = $this->pagedata->page_metadata('userviewself', 'menu');
+        $edittext   = $menudata['text'];
+        $edittitle  = $menudata['title'];
+        $EDITURL    = new \URL('userviewself');
+        if ($this->page == 'userviewself' || $this->page == 'useredit' || $this->header->top_highlight == 'userviewself') {
```

Comment:
Interestingly, I don't think the top_highlight can ever be userviewself (the only thing with it as a parent is useralerts which doesn't appear to be used anywhere…? Might have missed something) so this (and the passing of header into this) could hopefully be got rid of entirely.